metadata-io/src/main/java/com/linkedin/metadata/search/aggregator/AllEntitiesSearchAggregator.java

```diff
@@ -171,7 +171,7 @@ private Map<String, SearchResult> getSearchResultsForEachEntity(@Nonnull List<St
     // Query the entity search service for all entities asynchronously
     try (Timer.Context ignored = MetricUtils.timer(this.getClass(), "searchEntities").time()) {
       searchResults = ConcurrencyUtils.transformAndCollectAsync(entities, entity -> new Pair<>(entity,
-          _cachingEntitySearchService.search(entity, input, postFilters, sortCriterion, queryFrom, querySize, searchFlags, facets)))
+          _cachingEntitySearchService.search(List.of(entity), input, postFilters, sortCriterion, queryFrom, querySize, searchFlags, facets)))
```

Comment:
This is the part that needs to be changed, this aggregator class should no longer be necessary. This will still fire off a query for every entity, just using a singleton list instead of entity name string.

Reply:
Yes, I am planning to do that in a separate PR, since this one already had a lot of files modified. Is this okay as a refactor PR to change the interface with no functional changes?